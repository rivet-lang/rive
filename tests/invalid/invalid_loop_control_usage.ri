func main() {
    {
        break; // invalid
    }
    {
        continue; // invalid
    }

    while {
        defer {
            break; // invalid
        }
    }

    while {
        defer {
            continue; // invalid
        }
    }

    while {
        defer {
            while {
                break; // valid
            }
        }
        defer {
            while {
                continue; // valid
            }
        }
    }
}
