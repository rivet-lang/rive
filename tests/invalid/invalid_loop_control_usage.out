tests/invalid/invalid_loop_control_usage.ri:3:9: error: `break` and `continue` expressions can only be used inside loops
    3 |         break; // invalid
      |         ^~~~~
tests/invalid/invalid_loop_control_usage.ri:6:9: error: `break` and `continue` expressions can only be used inside loops
    6 |         continue; // invalid
      |         ^~~~~~~~
tests/invalid/invalid_loop_control_usage.ri:11:13: error: `break` and `continue` expressions inside loop cannot be deferred
   11 |             break; // invalid
      |             ^~~~~
tests/invalid/invalid_loop_control_usage.ri:17:13: error: `break` and `continue` expressions inside loop cannot be deferred
   17 |             continue; // invalid
      |             ^~~~~~~~
rivet: error: could not compile module `invalid_loop_control_usage`, aborting due to 4 previous errors